compiler/src/org.graalvm.compiler.replacements/src/org/graalvm/compiler/replacements/StringLatin1Substitutions.java

```diff
@@ -78,20 +78,11 @@ public static int indexOf(byte[] value, int ch, int origFromIndex) {
     }
 
     @MethodSubstitution
-    public static int indexOf(byte[] source, int sourceCount, byte[] target, int targetCount, int origFromIndex) {
-        int fromIndex = origFromIndex;
-        if (injectBranchProbability(UNLIKELY_PROBABILITY, fromIndex >= sourceCount)) {
-            return (targetCount == 0 ? sourceCount : -1);
-        }
```

Comment:
Should we restore this check? If sourceCount = fromIndex and targetCount = 0, we want to return sourceCount.
This check is already performed in [String.java](https://github.com/graalvm/labs-openjdk-11/blob/master/src/java.base/share/classes/java/lang/String.java#L1711-L1713). However, if this method is called from somewhere other than String.java, we may need to put this check.

Reply:
I left in two of the dynamic asserts, fromIndex >=0 and targetCount > 0, which I think must be true; otherwise the code would throw array out of bounds errors (which I imagine it doesn't)

Given these two assumptions, I think fromIndex >= sourceCount is unnecessary, as the scenario you pointed out can't happen; I think (sourceCount - fromIndex < targetCount) should be good enough